recipes/tls-create-secure-pair-to-tls-socket/src/workflow.ts

```diff
@@ -0,0 +1,236 @@
+			all: [
+				{ has: { kind: 'import_clause', has: { kind: 'named_imports' } } },
+				{
+					has: {
+						field: 'source',
+						any: [{ pattern: '\'tls\'' }, { pattern: '"tls"' }, { pattern: '\'node:tls\'' }, { pattern: '"node:tls"' }],
+					},
+				},
+			],
+			not: { has: { kind: 'namespace_import' } },
+		},
+	});
+	for (const decl of esmNamed) {
+		const srcText = decl.field('source')?.text()?.replace(/['"]/g, '') || '';
+		if (srcText !== 'tls' && srcText !== 'node:tls') continue;
+		const named = decl.find({ rule: { kind: 'named_imports' } });
+		if (!named) continue;
+		const specs = named.findAll({ rule: { kind: 'import_specifier' } });
+		const hasCSP = specs.some(s => s.field('name')?.text() === 'createSecurePair');
+		if (!hasCSP) continue;
+		const kept: string[] = [];
+		for (const s of specs) {
+			const imported = s.field('name')?.text();
+			const maybeAlias = s.field('alias')?.text();
+			if (imported && imported !== 'createSecurePair') {
+				kept.push(maybeAlias ? `${imported} as ${maybeAlias}` : imported);
+			}
+		}
+		if (kept.indexOf('TLSSocket') === -1) kept.push('TLSSocket');
+		const def = decl.find({ rule: { kind: 'import_clause', has: { field: 'name', kind: 'identifier' } } })?.field('name')?.text();
+		const rebuilt = def ? `import ${def}, { ${kept.join(', ')} } from '${srcText}';` : `import { ${kept.join(', ')} } from '${srcText}';`;
+		edits.push(decl.replace(rebuilt));
+	}
+	const cjsNamed = rootNode.findAll({
+		rule: {
+			kind: 'variable_declarator',
+			all: [
+				{ has: { field: 'name', kind: 'object_pattern' } },
+				{
+					has: {
+						field: 'value',
+						any: [
+							{ pattern: 'require(\'tls\')' },
+							{ pattern: 'require("tls")' },
+							{ pattern: 'require(\'node:tls\')' },
+							{ pattern: 'require("node:tls")' },
+						],
+					},
+				},
+			],
+		},
+	});
+	for (const decl of cjsNamed) {
+		const obj = decl.field('name');
+		if (!obj) continue;
+		const props = obj.findAll({ rule: { any: [{ kind: 'pair_pattern' }, { kind: 'shorthand_property_identifier_pattern' }] } });
+		const hasCSP = props.some(p =>
+			p.kind() === 'pair_pattern'
+				? p.field('key')?.text() === 'createSecurePair' || p.field('value')?.text() === 'createSecurePair'
+				: p.text() === 'createSecurePair'
+		);
+		if (!hasCSP) continue;
+		const kept: string[] = [];
+		for (const p of props) {
+			let name = '';
+			let alias = '';
+			if (p.kind() === 'pair_pattern') {
+				name = p.field('key')?.text() || '';
+				alias = p.field('value')?.text() || '';
+			} else {
+				name = p.text();
+				alias = name;
+			}
+			if (!name || name === 'createSecurePair') continue;
+			kept.push(alias && alias !== name ? `${name}: ${alias}` : name);
+		}
+		if (kept.indexOf('TLSSocket') === -1) kept.push('TLSSocket');
+		let stmt: SgNode<JS> = decl;
+		let cur: SgNode<JS> | undefined = decl;
+		while (cur) {
+			const k = cur.kind();
+			if (k === 'lexical_declaration' || k === 'variable_declaration') {
```

Comment:
Single letter var names usually indicate an index/counter. We're not short on space, so I think better to name it `kind`.

```suggestion
			const kind = cur.kind();
			if (kind === 'lexical_declaration' || kind === 'variable_declaration') {
```